### DISCUSSION TITLE
Open all programs button on each submenu

### DISCUSSION THREAD

**Discussion by BanCrash** [Hofknecht/SystemTrayMenu#382] [Q&A] [answered]
Hi, I think that on each submenu having a "Open all programs" button would be useful, so when you press it it will launch all programs on that submenu.

This will make that when you want to open all the programs of one category for any reason you don't have to click on one by one. 

In my idea this icon would be placed after the icon of launching the folder.

I just discovered this project and I love it, great work!

**Comment by Hofknecht** [comment]
@BanCrash 
Thanks for this idea!
The more I think about it, the more I like the idea. I think that can be implemented. I'm still thinking about whether a message should be displayed, since there could be many programs and the user might try it out unintentionally and then nothing should happen that surprises the user, but in principle it can be very practical!

I'll come back to this topic again.

**Reply by BanCrash** [reply]
> @BanCrash Thanks for this idea! The more I think about it, the more I like the idea. I think that can be implemented. I'm still thinking about whether a message should be displayed, since there could be many programs and the user might try it out unintentionally and then nothing should happen that surprises the user, but in principle it can be very practical!
> 
> I'll come back to this topic again.

Maybe the first time showing that message and adding the typical checkbox of "Don't show this message again" if the user just want to use it without double check all the times. I think this would be the better solution for both scenarios.

**Comment by Hofknecht** [comment, marked as answer]
@BanCrash
We thought about it again and think it's better to create this button yourself via bat file.
in this way you can control it precisely and e.g. also create a link and move it to the main menu.

create file 'launch my favorite apps.bat'
```
@ECHO OFF
start C:\Users\Marku\AppData\Local\Microsoft\Teams\Update.exe --processStart "Teams.exe"
start chrome
start "" "C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\devenv.exe"
start "" "D:\Seafile\Markus\C#Projects\systemtraymenu\SystemTrayMenu.sln"
```

get the target paths via right click on a item -> Properties

Sources:
https://stackoverflow.com/questions/5534324/how-to-run-multiple-programs-using-batch-file
https://www.makeuseof.com/tag/launch-multiple-programs-single-shortcut-using-batch-file/

**Reply by BanCrash** [reply]
Okay, thanks for the explanation!
